### PR TITLE
Top level interface connection

### PR DESCRIPTION
**Pull request by angrykoala**
# Description

Support for top-level connection query on interfaces. For example:

_Typedefs_

```graphql
interface Show {
    title: String!
}

type Movie implements Show {
    title: String!
    cost: Float
}

type Series implements Show {
    title: String!
    episodes: Int
}
```

_Query_

```graphql
query {
    showsConnection(where: { title_CONTAINS: "The Matrix" }) {
        edges {
            node {
                title
                ... on Movie {
                    cost
                }
            }
        }
    }
}
```

### CODE REVIEW

File: packages/graphql/src/schema-model/entity/model-adapters/InterfaceEntityAdapter.ts [hunk 1]
```diff
@@ -53,44 +53,11 @@ export class InterfaceEntityAdapter {
         this.initConcreteEntities(entity.concreteEntities);
     }
 
-    public findAttribute(name: string): AttributeAdapter | undefined {
```
Comment (angrykoala) [inline packages/graphql/src/schema-model/entity/model-adapters/InterfaceEntityAdapter.ts:56]: This bit here is just reordering some methods in InterfaceEntityAdapter to make it more consistent

**Comment by neo4j-team-graphql** [comment]
Thanks for the Neo4j GraphQL Toolbox updates.

The Neo4j GraphQL Toolbox has now been torn down - reopening this PR will republish it.

**Review by a-alle — DISMISSED** [review]
LGTM, maybe some tck tests could be useful?

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3404 | 3404 | 33 | 48 | 1134 |
|  aggregations.TopLevelAggregateWithMultipleFields | 6802 | 6802 | 45 | 70 | 1134 |
|  aggregations.NestedAggregation | 16544 | 16544 | 75 | 73 | 2174 |
|  aggregations.AggregationWithWhere | 11970 | 11970 | 47 | 58 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116478 | 22116478 | 2018 | 2175 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116478 | 22116478 | 2123 | 2023 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9701 | 9701 | 31 | 40 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9928 | 9928 | 34 | 38 | 2174 |
|  aggregations.DeeplyNestedCount | 11062184 | 11062184 | 2813 | 2769 | 2008534 |
|  aggregations.InterfacesAggregations | 6242 | 6242 | 45 | 47 | 2080 |
|  aggregations.InterfacesAggregationsWithTwoFields | 11444 | 11444 | 72 | 82 | 2080 |
|  batch-create.BatchCreate | 4200 | 4200 | 120 | 144 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 92 | 62 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 179 | 282 | 3003 |
|  connections.Connection | 14049 | 14049 | 77 | 81 | 2174 |
|  connections.NestedConnection | 41068 | 41068 | 117 | 148 | 4516 |
|  create.SimpleMutation | 7 | 7 | 48 | 58 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 57 | 26 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 228 | 205 | 1040 |
|  delete.NestedDeleteInUpdate | 19165 | 19165 | 165 | 202 | 2040 |
|  2871.NestedRelationshipFilter | 23083 | 23077 | 92 | 62 | 4431 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 44 | 44 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22891 | 22891 | 69 | 89 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 34 | 39 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 53 | 66 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 19 | 19 | 1040 |
|  query.SimpleQueryWithRelationship | 16129 | 16129 | 36 | 44 | 2174 |
|  query.QueryWhere | 9676 | 9676 | 31 | 34 | 2161 |
|  query.SimpleQueryWithNestedWhere | 9843 | 9843 | 45 | 49 | 2161 |
|  query.Nested | 10095502 | 10095502 | 7348 | 7079 | 2008534 |
|  query.NestedWithFilter | 10074351 | 10074351 | 6934 | 6866 | 2004000 |
|  query.OrFilterOnRelationships | 42814 | 43498 | 138 | 172 | 1750 |
|  query.OrFilterOnRelationshipsAndNested | 36073 | 36973 | 188 | 263 | 1750 |
|  query.QueryWithNestedIn | 13454 | 14010 | 49 | 55 | 1245 |
|  query.NestedConnectionWhere | 9801 | 9801 | 57 | 62 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9859 | 9799 | 71 | 84 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 18764 | 18686 | 127 | 159 | 1586 |
|  query.NestedWithRelationshipSingleFilters | 3865 | 3865 | 166 | 168 | 1134 |
|  query.Fulltext | 64 | 64 | 28 | 40 | 16 |
|  query.FulltextWithNestedQuery | 558 | 558 | 39 | 56 | 84 |
|  sorting-and-cypher.TopLevelSortWithCypher | 14059 | 14059 | 38 | 45 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypher | 14059 | 14059 | 61 | 73 | 2174 |
|  sorting-and-cypher.TopLevelSortWithCypherWithNested | 14204 | 14204 | 52 | 58 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypherWithNested | 14204 | 14204 | 105 | 104 | 2174 |
|  sorting-and-cypher.TopLevelSortWithExpensiveCypher | 14933 | 14922 | 93 | 126 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithExpensiveCypher | 14404 | 14404 | 101 | 125 | 2174 |
|  sorting.SortMultipleTypes | 3498 | 3498 | 89 | 96 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14504 | 14504 | 99 | 114 | 2174 |
|  sorting.SortOnNestedFields | 14049 | 14049 | 45 | 47 | 2174 |
|  sorting.SortDeeplyNestedFields | 43148 | 43148 | 75 | 90 | 4516 |
|  sorting.ConnectionWithSort | 3281 | 3281 | 86 | 89 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 64 | 57 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 92 | 56 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 250 | 334 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 256 | 239 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 101 | 103 | 2002 |

</details>
        
Old Schema Generation: 26.154s
Schema Generation: 26.345s
Old Subgraph Schema Generation: 30.524s
Subgraph Schema Generation: 30.297s

**Pull request merged**